src/main/java/io/confluent/idesidecar/restapi/featureflags/FeatureFlags.java

```diff
@@ -0,0 +1,738 @@
+            allProjects,
+            defaults.size(),
+            overrides.size()
+        );
+
+        // Create the latch that will be completed when all projects have refreshed
+        var latch = new CountDownLatch(deviceProjects.size() + ccloudProjects.size());
+
+        // Evaluate the project(s) using the device context.
+        // The cached evaluations are updated asynchronously.
+        deviceProjects.forEach(
+            project -> project.refresh(deviceContext(), latch)
+        );
+
+        // Evaluate the project(s) using the CCloud context.
+        // The cached evaluations are updated asynchronously.
+        ccloudProjects.forEach(
+            project -> project.refresh(ccloudContext(), latch)
+        );
+
+        // Set the next minimum time the flags CAN be run, using "almost" the interval
+        doNotRunUntil = now.plus(scheduledInterval);
+        if (scheduledInterval.isPositive()) {
+          Log.infof(
+              "Next evaluation of feature flags will be after %s (in %ss)",
+              HOUR_MINUTE.format(doNotRunUntil),
+              scheduledInterval.getSeconds()
+          );
+        }
+
+        // And finally return the latch that will complete when the refreshes do
+        return latch;
+      } else {
+        Log.infof(
+            "Skipping evaluation of feature flags until %s",
+            HOUR_MINUTE.format(doNotRunUntil)
+        );
+      }
+    }
+    return ZERO_LATCH;
+  }
+
+  /**
+   * Respond to the connection being disconnected by updating the LD context
+   * and re-evaluating the flags.
+   *
+   * <p>This updates the LaunchDarkly context used to evaluate feature flags to contain:
+   * <ol>
+   *   <li>
+   *     the general {@code Device} context with the {@link #DEVICE_UUID generated device UUID}; and
+   *   </li>
+   *   <li>
+   *     a {@code User} context with the logged in CCloud {@code userResourceId} as the context key
+   *   </li>
+   * </ol>
+   *
+   * @param connection the connection that was disconnected
+   */
+  void onConnectionConnected(@ObservesAsync @Lifecycle.Connected ConnectionState connection) {
+    if (connection instanceof CCloudConnectionState ccloudConnection) {
+      updateCCloudContext(ccloudConnection);
+    }
+    refreshFlags(true);
+  }
+
+  /**
+   * Respond to the connection being disconnected by updating the LD context
+   * and re-evaluating the flags.
+   *
+   * <p>This updates the LaunchDarkly context used to evaluate feature flags to contain only:
+   * <ol>
+   *   <li>
+   *     the general {@code Device} context with the {@link #DEVICE_UUID generated device UUID}; and
+   *   </li>
+   * </ol>
+   *
+   * @param connection the connection that was disconnected
+   */
+  void onConnectionDisconnected(@ObservesAsync @Lifecycle.Disconnected ConnectionState connection) {
+    // Set the context to the general context only
+    updateCCloudContext(null);
+    refreshFlags(true);
+  }
```

Comment:
How would this work if the extension supports multiple (connected) CCloud connections?

Reply:
Yeah, that's a big challenge that I chose to not address here, because I think there's no good solution as the feature flags almost certainly will not be evaluated within the context of a specific connection ID. 
IOW, as implemented, the feature flag evaluations will always reflect the most-recently connected CCloud connection.

I guess I justified this behavior with the current extension UX that has at most one CCloud connection, and that we'll probably stick with that for the foreseeable future.

Reply:
Makes sense! I also don't think that adding support for multiple CCloud connections justifies the effort here. I just want us to keep in mind that adding support for multiple CCloud connections will require a bigger effort on the sidecar side after merging this PR.